WooCommerce/Classes/ViewRelated/Orders/Order Creation/Synchronizer/RemoteOrderSynchronizer.swift

```diff
@@ -206,10 +208,22 @@ private extension RemoteOrderSynchronizer {
             }
             .store(in: &subscriptions)
 
-        setCoupon.withLatestFrom(orderPublisher)
+        addCoupon.withLatestFrom(orderPublisher)
             .map { couponLineInput, order -> Order in
-                let updatedOrder = CouponInputTransformer.update(input: couponLineInput, on: order)
-                return updatedOrder
+                let updatedOrder = CouponInputTransformer.append(input: couponLineInput, on: order)
+                // Calculate order total locally while order is being synced
+                return OrderTotalsCalculator(for: updatedOrder, using: self.currencyFormatter).updateOrderTotal()
+            }
+            .sink { [weak self] order in
+                self?.order = order
+                self?.orderSyncTrigger.send(order)
+            }
+            .store(in: &subscriptions)
+        removeCoupon.withLatestFrom(orderPublisher)
+            .map { couponCode, order -> Order in
```

Comment:
I'm not 100% sure about this, but should we also weakify the reference to the order here? We seem to do it inconsistently through the different binded inputs despite all of them being declared performing similar actions:

```suggestion
            .map { [weak self] couponCode, order -> Order in
                guard let self = self else { return order }
```

Same for `addCoupon.withLatestFrom(orderPublisher).map { ...`

Reply:
Yeah, it was like that before, but just in case I added it in 4455d3eb8b4e48a82fbc525731fea0341c00db8c